pages/DSNP/UserData.md

```diff
@@ -0,0 +1,211 @@
+
+The Replace User Data Operation takes the following parameters:
+
+* A DSNP User Id
+  * Implementations MUST ensure that the principal invoking this Operation is this user, or a transparent chain of delegation from the user to the principal exists.
+* A [Key Identifier](Types/PublicKey.md#keyid) for the `keyAgreement` key pair used to encrypt any private data in the operation.
+  (If only unencrypted user data types are included, the key identifier is optional.)
+* A map containing the set of data types to update as the keys, and tuples consisting of (1) the schema version used to encode the data type, and (2) a list where each element includes a data chunk and its associated entity tag, as the values.
+
+If the Operation is successful, any previous data associated with the user for each data type included in the input MUST be removed and replaced by the new data.
+
+Data chunks should be generated for each included data type using the following sequence of operations:
+1. Serialize the data records, in Avro binary format, according to the versioned schema.
+1. Divide the data records into one or more chunks, with each chunk containing a maximum byte count specified by the implementation.
+* Note that the combined chunks constitute a different, and smaller, binary than a complete Avro file, as they exclude the header information, schema, and any block-level metadata.
+  Each chunk should start and end with a complete Avro object; in other words, chunk boundaries MUST NOT occur mid-record.
+    Applications SHOULD try to include as many records as possible within the byte length limit, but because compression and encryption may alter the byte size versus the raw Avro binary records, they may wish to use a heuristic approach that sometimes produces non-optimal chunks.
+3. For each chunk generated, the application should then:
+    1. If the data type requires compression, apply the compression codec noted.
+    1. If the data type requires encryption,
+        1. Retrieve the user's active (most recently announced) `keyAgreement` public key, U<sub>public</sub>.
+  The `keyId` in the announcement should match the key identifier provided for this Operation.
+  If no key exists, one should be created and published as an Announcement before invoking the Operation.
+        1. Create a sealed box, as in the [libsodium](https://doc.libsodium.org/public-key_cryptography/sealed_boxes) function `crypto_box_seal`, using U<sub>public</sub>.
+        1. Include the previous `etag` value for the chunk. If the chunk is new, `etag` should be set to `null`.
+  If any chunks are to be deleted, they should be included in the input identified with the existing `etag` and a `null` value for the data.
+
+If the Operation is invoked successfully the implementation MUST (synchronously) return a new set of `etag` values for each data type replaced, corresponding to the updated state of the data (with new chunks added and deleted chunks removed).
```

Comment:
If the Operation is successful is a synchronous return actually required? If so, "synchronously" being in parens is a little odd.

Reply:
```suggestion
If the Operation is invoked successfully the implementation MUST synchronously return a new set of `etag` values for each data type replaced, corresponding to the updated state of the data (with new chunks added and deleted chunks removed).
```